pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -96,13 +97,22 @@ func (alloc *Action) Execute(ssn *framework.Session) {
 	pendingTasks := map[api.JobID]*util.PriorityQueue{}
 
 	allNodes := ssn.NodeList
-	predicateFn := func(task *api.TaskInfo, node *api.NodeInfo) error {
+	predicateFn := func(task *api.TaskInfo, node *api.NodeInfo) ([]*api.Status, error) {
 		// Check for Resource Predicate
 		if ok, reason := task.InitResreq.LessEqualWithReason(node.FutureIdle(), api.Zero); !ok {
-			return api.NewFitError(task, node, reason)
+			return nil, api.NewFitError(task, node, reason)
+		}
+		predicateStatus, err := ssn.PredicateFn(task, node)
```

Comment:
Thank you for your review， here is mainly the core logic that distinguishes the difference between allocate and preempt. It has been modified using the previous logical architecture. 

I will consider it later to see if it can be further optimized.